src/spatial/detail/ArborX_SpaceFillingCurves.hpp

```diff
@@ -79,6 +61,50 @@ struct Morton64
 namespace Details
 {
 
+template <typename ExecutionSpace, typename Values, typename SpaceFillingCurve,
+          typename Box>
+inline Kokkos::View<std::invoke_result_t<SpaceFillingCurve, Box,
+                                         std::decay_t<decltype(returnCentroid(
+                                             std::declval<Values>()(0)))>> *,
+                    typename Values::memory_space>
+projectOntoSpaceFillingCurve(ExecutionSpace const &space, Values const &values,
+                             SpaceFillingCurve const &curve,
+                             Box const &scene_bounding_box)
+{
+  using Point = std::decay_t<decltype(returnCentroid(values(0)))>;
+  static_assert(GeometryTraits::is_point_v<Point>);
+  static_assert(GeometryTraits::is_box_v<Box>);
+
+  auto const n = values.size();
+
+  using LinearOrderingValueType =
+      std::invoke_result_t<SpaceFillingCurve, Box, Point>;
+  Kokkos::View<LinearOrderingValueType *, typename Values::memory_space>
+      linear_ordering_indices(
+          Kokkos::view_alloc(space, Kokkos::WithoutInitializing,
+                             "ArborX::SpaceFillingCurve::linear_ordering"),
+          n);
```

Comment:
I don't think that moving the allocation into that function and returning the Morton codes is a good change

Reply:
I don't like it much either. However, our boilerplate code for determining the type of indices is really several lines at every call site.

Alternatively, we could introduce a requirement for a space filling curve to provide `index_type` that it will return. It would simplify things a lot.